File: codegen/results/v002/qwen2.5-coder_7b/20260411_224712/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75

# Test cases
test_cases = [
    (10, 5, "quantity below threshold"),
    (20, 10, "quantity at lower boundary of discount"),
    (30, 49, "quantity within discount range"),
    (40, 50, "quantity at upper boundary of discount")
]

for price, quantity, description in test_cases:
    expected = calculate_discount(price, quantity)
    result = calculate_discount(price, quantity)
    if expected == result:
        print(f"PASS: {description}")
    else:
        print(f"FAIL: {description} (expected: {expected}, got: {result})")

File: codegen/results/v002/qwen2.5-coder_7b/20260411_224712/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Test cases
const testCases = [
    { price: 10, quantity: 9, expected: 90 },
    { price: 10, quantity: 10, expected: 100 },
    { price: 10, quantity: 49, expected: 441 },
    { price: 10, quantity: 50, expected: 375 }
];

for (const testCase of testCases) {
    const result = calculateDiscount(testCase.price, testCase.quantity);
    if (result === testCase.expected) {
        console.log(`PASS: Expected discount for price ${testCase.price} and quantity ${testCase.quantity} is ${testCase.expected}`);
    } else {
        console.log(`FAIL: Expected discount for price ${testCase.price} and quantity ${testCase.quantity} to be ${testCase.expected}, but got ${result}`);
    }
}